tools/build_updated_packages.rb

```diff
@@ -82,11 +82,27 @@ def update_hashes(name = nil)
     # Add build hashes.
     system "crew update_package_file #{name}" unless remote_binary.values.all?(nil)
     # Add manifests if we are in the right architecture.
-    # Using crew reinstall -f package here updates the hashes for
-    # binaries.
-    if system("yes | crew reinstall --regenerate-filelist #{'-f' unless CREW_BUILD_NO_PACKAGE_FILE_HASH_UPDATES} #{name}") && File.exist?("#{CREW_META_PATH}/#{name}.filelist") && File.directory?(CREW_LOCAL_REPO_ROOT)
-      puts 'Adding manifests...'
-      FileUtils.cp "#{CREW_META_PATH}/#{name}.filelist", "#{CREW_LOCAL_REPO_ROOT}/manifest/#{ARCH}/#{name.chr}/#{name}.filelist"
+    if @pkg_obj.compatibility == 'all' || @pkg_obj.compatibility.include?(ARCH)
+      # Using crew reinstall -f package here updates the hashes for
+      # binaries.
+      if system("yes | crew reinstall --regenerate-filelist #{'-f' unless CREW_BUILD_NO_PACKAGE_FILE_HASH_UPDATES} #{name}") && File.exist?("#{CREW_META_PATH}/#{name}.filelist") && File.directory?(CREW_LOCAL_REPO_ROOT)
+        puts 'Adding manifests...'
+        FileUtils.cp "#{CREW_META_PATH}/#{name}.filelist", "#{CREW_LOCAL_REPO_ROOT}/manifest/#{ARCH}/#{name.chr}/#{name}.filelist"
+      end
+    else
+      puts "Package #{name} is not compatible with your device architecture (#{ARCH}). Manifests will not be added.".orange
```

Comment:
Is this a warning or error?  If error, shouldn't it be red instead of orange?

Reply:
It's more informational than a warning. It's expected that you can't get manifests from a package if you aren't in in that architecture.